NuRadioReco/modules/io/RNO_G/readRNOGDataMattak.py

```diff
@@ -308,9 +306,13 @@ def begin(self,
             
                 if not all_files_in_directory(dir_file):
                     self.logger.error(f"Incomplete directory: {dir_file}. Skip ...")
-                    continue      
-            
-                dataset = mattak.Dataset.Dataset(station=0, run=0, data_dir=dir_file, verbose=verbose, **mattak_kwargs)
+                    continue
+                
+                try:
+                    dataset = mattak.Dataset.Dataset(station=0, run=0, data_dir=dir_file, verbose=verbose, **mattak_kwargs)
+                except Exception as e:
+                    self.logger.error(f"The following Exeption was raised reading in the run: f{dir_file}. Skip that run ...: {e}")
+                    continue
```

Comment:
This is actually not so easy because it depends on the backend: for uproot we have `uproot.exceptions.KeyInFileError`, for pyroot we have a `ReferenceError`. And I could imagine that this changes depending on how the files are broken

Reply:
Is uproot a fixed dependency? Then we can start with the two errors above but we have to be aware that other problems might occur.

Reply:
uproot.exceptions.KeyInFileError inherits from `KeyError`, so it would be sufficient to catch `(KeyError, ReferenceError)`. Although it may be worth thinking about raising a single error in mattak.

Having said that, I've now done my homework, and it seems a `KeyboardInterrupt` would not be caught by `Exception` (it's a `BaseException` specifically for that reason). So I'm a bit more torn now; if we're really not that sure which errors might occur (and assuming that we don't want `readRNOGData` to raise an error in any of those cases) there is some value in leaving it the way you've implemented it now. I might leave the decision up to you.

Reply:
arg, I saw your msg to late